packages/openapi-generator/src/parser/schema.ts

```diff
@@ -221,8 +223,34 @@ function parseXOfSchema(
   xOf: 'oneOf' | 'allOf' | 'anyOf',
   options: ParserOptions
 ): any {
+  let normalizedSchema: OpenAPIV3.NonArraySchemaObject = schema;
+  if (
+    schema.properties ||
+    (schema.additionalProperties &&
+      typeof schema.additionalProperties === 'object')
+  ) {
+    logger.info(
+      `Detected schema with ${xOf} and properties in the same level. This was refactored to a schema with ${xOf} only, containing all the properties from the top level.`
+    );
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { [xOf]: _, ...objectSchema } = schema;
+
+    normalizedSchema = { [xOf]: [...(schema[xOf] || []), objectSchema] };
+  }
   return {
-    [xOf]: (schema[xOf] || []).map(entry => parseSchema(entry, refs, options))
+    [xOf]: (normalizedSchema[xOf] || []).map(entry =>
+      parseSchema(
+        {
+          ...entry,
+          required: [
+            ...('required' in entry && entry.required ? entry.required : []),
```

Comment:
[q] Why is this necessary? I would have expected that this is enough:
```suggestion
            ...(entry.required || []),
```

Reply:
required does not exist on type 'ReferenceObject'. But I guess I can also rewrite it as :

`...(entry['required'] || []),`